### DISCUSSION TITLE
"Anonymous arrow functions cause Fast Refresh" warning triggered by object, not function

### DISCUSSION THREAD

**Discussion by YuriGor** [vercel/next.js#32382] [Help]
### What version of Next.js are you using?

11.1.0

### What version of Node.js are you using?

14.17.0

### What browser are you using?

Chrome

### What operating system are you using?

Ubuntu

### How are you deploying your application?

Vercel

### Describe the Bug

I am using next.j + Material UI
For Material UI I have theme.js:
```js
import { createTheme } from '@material-ui/core/styles';
import { themeColors } from './colors';

// Create a theme instance.
const theme = createTheme({
  components: {
    MuiTextField: {
      defaultProps: { variant: 'standard' },
    },
    MuiDatePicker: {
      defaultProps: { variant: 'standard' },
    },
    MuiSelect: {
      variant: 'standard',
    },
  },
  transition: {
    duration: '0.3s',
  },
  palette: {
    primary: {
      main: themeColors.primary,
    },
    secondary: {
      main: themeColors.secondary,
    },
    error: {
      main: themeColors.error,
    },
    background: {
      default: themeColors.background,
    },
    common: {
      grey: themeColors.common,
    },
  },
});

console.log(`theme is ${typeof theme}`, theme);

export default theme;
```
When compiling page I see in CLI:
```
Vercel CLI 23.1.2 dev (beta) — https://vercel.com/feedback
> Running Dev Command “next dev --port $PORT”
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Loaded env from /my/cool/project/.env.local
info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
> Ready! Available at http://localhost:3000
info  - automatically enabled Fast Refresh for 1 custom loader
warn  - ./src/theme.js
Anonymous arrow functions cause Fast Refresh to not preserve local component state.
Please add a name to your function, for example:

Before
export default () => <div />;

After
const Named = () => <div />;
export default Named;

A codemod is available to fix the most common cases: https://nextjs.link/codemod-ndc

```
console.log output from theme.js:
```
theme is object {
  breakpoints: {
    keys: [ 'xs', 'sm', 'md.................
```

### Expected Behavior

No warnings, because it's not a function.

### To Reproduce

Issue is not always reproducing, but this warning should check type of what was exported.

**Comment by manudeli** [comment]
I encounter this problem too.

is There any solution for this problem yet?

**Reply by YuriGor** [reply]
For me, when it appears, seems to help stop, delete .next/cache, ,vercel/cache (I use vercel cli for dev) and not sure but probably node_modules and npm install everything again.

**Reply by manudeli** [reply]
@YuriGor Please check this PR 😎

This PR(#32763) may proper answer for your discussion!
You and I may need to upgrade version of Next.js to 12.0.8 (latest version)

**Reply by YuriGor** [reply]
![image](https://user-images.githubusercontent.com/6861317/150657512-e86ec08e-42fc-46eb-a0c2-35c72194f943.png)
em.. ok, thanks, maybe _next_ time
